components/leave_team_modal/leave_team_modal.tsx

```diff
@@ -4,52 +4,24 @@
 import React from 'react';
 import {Modal} from 'react-bootstrap';
 import {FormattedMessage} from 'react-intl';
-import PropTypes from 'prop-types';
+import {ActionFunc} from 'mattermost-redux/types/actions';
 
 import Constants from 'utils/constants';
 import {isKeyPressed} from 'utils/utils';
 
-export default class LeaveTeamModal extends React.PureComponent {
-    static propTypes = {
-
-        /**
-         * Current user id.
-         */
-        currentUserId: PropTypes.string.isRequired,
-
-        /**
-         * Current team id.
-         */
-        currentTeamId: PropTypes.string.isRequired,
-
-        /**
-         * hide action
-         */
-
-        onHide: PropTypes.func.isRequired,
-
-        /**
-         * show or hide modal
-         */
-
-        show: PropTypes.bool.isRequired,
-
-        actions: PropTypes.shape({
-
-            /**
-             * An action to remove user from team
-             */
-
-            leaveTeam: PropTypes.func.isRequired,
-
-            /**
-             * An action to toggle the right menu
-             */
-
-            toggleSideBarRightMenu: PropTypes.func.isRequired,
-        }),
+type Props = {
+    currentUserId: string;
+    currentTeamId: string;
+    onHide: () => {};
```

Comment:
Non-blocking, since this doesn't return anything this can return `void` instead of `{}`